tools/test.py

```diff
@@ -1757,10 +1760,8 @@ def should_keep(case):
   else:
     try:
       start = time.time()
-      if RunTestCases(cases_to_run, options.progress, options.j, options.flaky_tests, options.measure_flakiness):
-        result = 0
-      else:
-        result = 1
+      result = RunTestCases(cases_to_run, options.progress, options.j, options.flaky_tests, options.measure_flakiness)
+      exitCode = 0 if result['allPassed'] else 1
```

Comment:
```suggestion
      failed_tests = RunTestCases(cases_to_run, options.progress, options.j, options.flaky_tests, options.measure_flakiness)
      exit_code = int(bool(failed_tests))  # Or maybe `len(failed_tests)````

Reply:
Thank you for the suggestion but I think the dict return is the better approach.